src/road_cmd.cpp

```diff
@@ -1162,24 +1162,40 @@ CommandCost CmdBuildRoadDepot(DoCommandFlag flags, TileIndex tile, RoadType rt,
 		cost.AddCost(_price[PR_BUILD_FOUNDATION]);
 	}
 
-	cost.AddCost(Command<CMD_LANDSCAPE_CLEAR>::Do(flags, tile));
-	if (cost.Failed()) return cost;
+	/* Allow the user to rotate the depot instead of having to destroy it and build it again */
+	bool rotate_existing_depot = false;
+	if (IsRoadDepotTile(tile) && (HasRoadTypeTram(tile) ? rt == RoadType::ROADTYPE_TRAM : rt == RoadType::ROADTYPE_ROAD)) {
+		return_on_cmd_failure(CheckTileOwnership(tile));
+		if (dir == GetRoadDepotDirection(tile)) return_cmd_error(STR_ERROR_ALREADY_BUILT);
+		rotate_existing_depot = true;
+		return_on_cmd_failure(EnsureNoVehicleOnGround(tile));
```

Comment:
To match CmdBuildTrainDepot():

```suggestion
		return_on_cmd_failure(EnsureNoVehicleOnGround(tile));
		rotate_existing_depot = true;
```

Reply:
Nitpicks addressed as suggested ;)